services/users/src/permission.ts

```diff
@@ -2,8 +2,16 @@ import { AbilityAction } from "@api/common";
 import { AbilityBuilder, Ability, Subject } from "@casl/ability";
 import { RequestHandler } from "express";
 
-export const addAbilities = (): RequestHandler => (req, res, next) => {
+import { CompanyModel } from "./models/company";
+
+export const addAbilities = (): RequestHandler => async (req, res, next) => {
   const { can, build } = new AbilityBuilder<Ability<[AbilityAction, Subject]>>(Ability);
+
+  const isSponsorWithResumeAccess = async (user: any) => {
+    const company = await CompanyModel.findOne({ employees: user.uid });
+    return company?.hasResumeAccess;
+  };
+
```

Comment:
I don't think this will work, we'll need to move it over the endpoints specifically, because we have to account for the hexathon, as each hexathon will have its own set of companies, so a user could below to multiple companies for different events.

Reply:
Ooo yea you're right. 
Question: For companies without resume access, can they just only see basic info and not the resume?

Reply:
What I’ve done also is for other routes, always pass the hexathon as a query param (like after the ? In the url) and use that to get an employees company for a certain hexathon. Yeah, they can only see basic info not resume.

Reply:
Update: removed this logic from permission.ts to specific routes. My thought: we can query for accepted applications based on whichever hexathon, and then sponsors have permission to query a user by their id, but no permission to query all users (since a user isn't tied to just one specific hexathon, so we shouldn't let them potentially see users in hexathons they didn't sponsor). resume access logic should be moved to front end since sponsors should still see the basic information about a user.